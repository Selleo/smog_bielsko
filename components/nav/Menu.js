import React, { Component }                          from 'react';
import { Dimensions, ScrollView, View, Image, Text } from 'react-native';

import menu                                          from './../stylesheets/Menu'

const uri = 'https://pickaface.net/gallery/avatar/Opi51c74d0125fd4.png';
const window = Dimensions.get('window');

module.exports = class Menu extends Component {
  static propTypes = {
    onItemSelected: React.PropTypes.func.isRequired,
  };

  render() {
    return (
      <ScrollView scrollsToTop={false} style={menu.menu}>
        <View style={menu.avatarContainer}>
          <Image
            style={menu.avatar}
            source={{ uri, }}/>
          <Text style={menu.name}>Your name</Text>
        </View>

        <Text
          onPress={() => this.props.onItemSelected('About')}
          style={menu.item}>
          About
        </Text>

        <Text
          onPress={() => this.props.onItemSelected('Contacts')}
          style={menu.item}>
          Contacts
        </Text>
      </ScrollView>
    );
  }
};
