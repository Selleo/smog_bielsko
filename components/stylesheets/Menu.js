import StyleSheet from 'react-native-extensible-stylesheet'

export default StyleSheet.create({
  button: {
    position: 'absolute',
    top: 20,
    padding: 10,
  },
  caption: {
    fontSize: 20,
    fontWeight: 'bold',
    alignItems: 'center',
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
  menu: {
    flex: 1,
    width: window.width,
    height: window.height,
    backgroundColor: 'gray',
    padding: 20,
  },
  avatarContainer: {
    marginBottom: 20,
    marginTop: 20,
  },
  avatar: {
    width: 48,
    height: 48,
    borderRadius: 24,
    flex: 1,
  },
  name: {
    position: 'absolute',
    left: 70,
    top: 20,
  },
  item: {
    fontSize: 14,
    fontWeight: '300',
    paddingTop: 5,
  },
})
