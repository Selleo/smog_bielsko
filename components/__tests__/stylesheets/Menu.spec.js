import 'react-native';
import React                          from 'react';

import { shallow }                    from 'enzyme';
import chai, { expect }               from 'chai';
import dirtyChai                      from 'dirty-chai';

import Menu                           from '../../stylesheets/Menu';

const originalExpect = global.expect;

chai.use(dirtyChai);

describe('Menu', () => {
  it('renders correctly', () => {
    originalExpect(Menu).toMatchSnapshot();
    expect(Menu).to.exist();
  });
});
