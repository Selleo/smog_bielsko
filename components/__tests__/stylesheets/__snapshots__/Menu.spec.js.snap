exports[`Menu renders correctly 1`] = `
ExtensibleStyleSheet {
  "_options": Object {
    "avatar": Object {
      "borderRadius": 24,
      "flex": 1,
      "height": 48,
      "width": 48,
    },
    "avatarContainer": Object {
      "marginBottom": 20,
      "marginTop": 20,
    },
    "button": Object {
      "height": 32,
      "padding": 10,
      "position": "absolute",
      "top": 20,
    },
    "caption": Object {
      "alignItems": "center",
      "fontSize": 20,
      "fontWeight": "bold",
    },
    "container": Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    },
    "instructions": Object {
      "color": "#333333",
      "marginBottom": 5,
      "textAlign": "center",
    },
    "item": Object {
      "fontSize": 14,
      "fontWeight": "300",
      "paddingTop": 5,
    },
    "menu": Object {
      "backgroundColor": "gray",
      "flex": 1,
      "height": undefined,
      "padding": 20,
      "width": undefined,
    },
    "name": Object {
      "left": 70,
      "position": "absolute",
      "top": 20,
    },
    "welcome": Object {
      "fontSize": 20,
      "margin": 10,
      "textAlign": "center",
    },
  },
  "_styles": Object {
    "avatar": Object {
      "borderRadius": 24,
      "flex": 1,
      "height": 48,
      "width": 48,
    },
    "avatarContainer": Object {
      "marginBottom": 20,
      "marginTop": 20,
    },
    "button": Object {
      "height": 32,
      "padding": 10,
      "position": "absolute",
      "top": 20,
    },
    "caption": Object {
      "alignItems": "center",
      "fontSize": 20,
      "fontWeight": "bold",
    },
    "container": Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    },
    "instructions": Object {
      "color": "#333333",
      "marginBottom": 5,
      "textAlign": "center",
    },
    "item": Object {
      "fontSize": 14,
      "fontWeight": "300",
      "paddingTop": 5,
    },
    "menu": Object {
      "backgroundColor": "gray",
      "flex": 1,
      "height": undefined,
      "padding": 20,
      "width": undefined,
    },
    "name": Object {
      "left": 70,
      "position": "absolute",
      "top": 20,
    },
    "welcome": Object {
      "fontSize": 20,
      "margin": 10,
      "textAlign": "center",
    },
  },
  "_variables": Object {},
  "avatar": Object {
    "borderRadius": 24,
    "flex": 1,
    "height": 48,
    "width": 48,
  },
  "avatarContainer": Object {
    "marginBottom": 20,
    "marginTop": 20,
  },
  "button": Object {
    "height": 32,
    "padding": 10,
    "position": "absolute",
    "top": 20,
  },
  "caption": Object {
    "alignItems": "center",
    "fontSize": 20,
    "fontWeight": "bold",
  },
  "container": Object {
    "alignItems": "center",
    "backgroundColor": "#F5FCFF",
    "flex": 1,
    "justifyContent": "center",
  },
  "instructions": Object {
    "color": "#333333",
    "marginBottom": 5,
    "textAlign": "center",
  },
  "item": Object {
    "fontSize": 14,
    "fontWeight": "300",
    "paddingTop": 5,
  },
  "menu": Object {
    "backgroundColor": "gray",
    "flex": 1,
    "height": undefined,
    "padding": 20,
    "width": undefined,
  },
  "name": Object {
    "left": 70,
    "position": "absolute",
    "top": 20,
  },
  "welcome": Object {
    "fontSize": 20,
    "margin": 10,
    "textAlign": "center",
  },
}
`;
